[ops]: ttnn::std PCC failures for multi-dimensional reductions with use_legacy=True

### Component / Area

ops

### Issue Type (optional)

Bad Outputs

### Observed

Test: 
`tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py::test_generic_ops`

While running the above test, PCC failures were observed for the **std** operation on certain tensor shapes when reducing across multiple dimensions (for example: `dim = (1, 2, 3)`). The failures are observed with **BF16 inputs** when **use_legacy=True** and are listed in the table below. (**pcc_threshold = 0.98**)

### Failing Cases

| Shape | Dim | Op | Keepdim | Correction | Use Legacy | Status | PCC |
|---------|---------|---------|---------|---------|---------|---------|---------|
| (4, 8, 32, 64) | (0, 2, 3) | std | False | True | True | FAIL | 0.9201231009113796 |
| (3, 6, 40, 63, 20) | (1, 2, 3) | std | False | True | True | FAIL | 0.9682608623256354 |

### Expected



The **std** operation should meet the required **PCC threshold of 0.98** for BF16 inputs when performing multi-dimensional reductions with `use_legacy=True`. All listed test cases are expected to pass.

### 1. Steps (exact commands)

1. Include input shape and dim of above failing test cases (if not exist) in                                                                   `tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py::test_generic_ops`

2. Run the below command
`pytest \
'tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py::test_generic_ops[use_legacy=True-op=std-correction=True-layout=Layout.TILE-dtype=torch.bfloat16-keepdim=False-dim=(0, 2, 3)-tensor_shape=(4, 8, 32, 64)]' \
'tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py::test_generic_ops[use_legacy=True-op=std-correction=True-layout=Layout.TILE-dtype=torch.bfloat16-keepdim=False-dim=(1, 2, 3)-tensor_shape=(3, 6, 40, 63, 20)]'`

### 2. Input data / link or description

Input data is generated within the test using a fixed random seed. No external datasets or files are required.

```python
torch.manual_seed(0)
torch_tensor = torch.randn(tensor_shape, dtype=torch.bfloat16)
```

### 3. Frequency

Always

### 1. Software Versions

Ubuntu 22.04

### 2. Hardware Details

Wormhole (N150)

### Is this a regression?

Unknown

### Regression Details

_No response_

### Logs & Diagnostics

```shell
FAILED tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py::test_generic_ops[use_legacy=True-op=std-correction=True-layout=Layout.TILE-dtype=torch.bfloat16-keepdim=False-dim=(0, 2, 3)-tensor_shape=(4, 8, 32, 64)] - AssertionError: Max ATOL Delta: 0.00390625, Max RTOL Delta: 0.003997802734375, pcc=0.9201231009113796, torch: tensor([0.9883, 0.9961, 0.9961, 0.9961, 0.9961, 0.9961, 0.9844, 0.9961],
         dtype=torch.bfloat16), ttnn: tensor([0.9883, 0.9922, 0.9922, 0.9922, 0.9961, 0.9961, 0.9805, 0.9961],
         dtype=torch.bfloat16)
assert False
FAILED tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py::test_generic_ops[use_legacy=True-op=std-correction=True-layout=Layout.TILE-dtype=torch.bfloat16-keepdim=False-dim=(1, 2, 3)-tensor_shape=(3, 6, 40, 63, 20)] - AssertionError: Max ATOL Delta: 0.0078125, Max RTOL Delta: 0.00799560546875, pcc=0.9682608623256354, torch: tensor([[0.9922, 0.9922, 0.9922, 0.9922, 0.9922, 1.0000, 0.9922, 1.0000, 1.0000,
           0.9922, 0.9922, 0.9883, 0.9961, 0.9883, 1.0078, 1.0000, 0.9961, 0.9961,
           0.9883, 0.9922],
          [0.9883, 0.9922, 0.9883, 1.0000, 0.9961, 1.0000, 0.9883, 0.9961, 0.9922,
           0.9922, 0.9922, 1.0000, 0.9805, 0.9922, 0.9766, 0.9922, 0.9922, 0.9922,
           0.9805, 0.9922],
          [1.0000, 0.9922, 0.9922, 0.9922, 0.9883, 0.9844, 0.9922, 0.9961, 0.9922,
           0.9961, 0.9961, 0.9883, 1.0000, 0.9844, 0.9844, 0.9961, 0.9883, 0.9961,
           1.0000, 0.9961]], dtype=torch.bfloat16), ttnn: tensor([[0.9883, 0.9883, 0.9883, 0.9883, 0.9883, 0.9961, 0.9883, 0.9961, 0.9961,
           0.9883, 0.9883, 0.9844, 0.9922, 0.9844, 1.0000, 0.9961, 0.9922, 0.9922,
           0.9844, 0.9883],
          [0.9844, 0.9883, 0.9844, 1.0000, 0.9922, 0.9922, 0.9844, 0.9922, 0.9883,
           0.9883, 0.9883, 1.0000, 0.9766, 0.9883, 0.9727, 0.9883, 0.9883, 0.9883,
           0.9766, 0.9883],
          [1.0000, 0.9883, 0.9883, 0.9883, 0.9844, 0.9805, 0.9883, 0.9883, 0.9883,
           0.9883, 0.9922, 0.9805, 0.9961, 0.9805, 0.9805, 0.9922, 0.9844, 0.9922,
           0.9961, 0.9922]], dtype=torch.bfloat16)
assert False
```

### Priority

None

### Impact

_No response_

--- a/tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py
+++ b/tests/ttnn/nightly/unit_tests/operations/reduction/test_reduction_ops.py
@@ -202,8 +202,7 @@ def _torch_sampling_reference(values, indices, k, p, temp, seed):
 @pytest.mark.parametrize("layout", [ttnn.TILE_LAYOUT])
 @pytest.mark.parametrize("correction", [True, False])
 @pytest.mark.parametrize("op", ["mean", "sum", "max", "min", "prod", "std", "var"])
-@pytest.mark.parametrize("use_legacy", [True, False])
-def test_generic_ops(device, tensor_shape, dim, keepdim, dtype, layout, correction, op, use_legacy):
+def test_generic_ops(device, tensor_shape, dim, keepdim, dtype, layout, correction, op):
     """
     Test the compatibility of the torch and ttnn output for the given operation and different
     tensor shapes, keepdim, and dim values.
@@ -211,9 +210,6 @@ def test_generic_ops(device, tensor_shape, dim, keepdim, dtype, layout, correcti
     Some operations raise exceptions in torch, we check if the same behavior is observed in ttnn.
     Note: We do not enforce the same exception type or message.
     """
-    if op not in ("var", "std") and use_legacy:
-        pytest.skip("use_legacy only applies to std and var")
-
     if op not in ("var", "std") and correction:
         pytest.skip("PyTorch supports the correction argument only for var and std")
 
@@ -259,7 +255,7 @@ def test_generic_ops(device, tensor_shape, dim, keepdim, dtype, layout, correcti
     try:
         # ttnn.prod doesn't support the correction argument.
         if op in ("var", "std"):
-            ttnn_result = ttnn_op(ttnn_tensor, dim=dim, keepdim=keepdim, correction=correction, use_legacy=use_legacy)
+            ttnn_result = ttnn_op(ttnn_tensor, dim=dim, keepdim=keepdim, correction=correction)
         elif op != "prod":
             ttnn_result = ttnn_op(ttnn_tensor, dim=dim, keepdim=keepdim, correction=correction)
         else:
@@ -290,10 +286,6 @@ def test_generic_ops(device, tensor_shape, dim, keepdim, dtype, layout, correcti
         # and results also vary from near 0 to relatively large values (in hundreds)
         # PCC should catch any significant errors.
         atol = 1.5
-    elif use_legacy and op == "std":
-        # Legacy two-pass method (E[X^2] - E[X]^2) suffers from more catastrophic cancellation
-        # than the Welford single-pass path, especially in bfloat16, so thresholds are slightly relaxed.
-        atol = 0.25
     else:
         atol = 0.1
 

--- a/tests/ttnn/unit_tests/operations/reduce/test_reduction.py
+++ b/tests/ttnn/unit_tests/operations/reduce/test_reduction.py
@@ -18,8 +18,7 @@ TEST_PADDING_VALUE = -42
 @pytest.mark.parametrize("dim", [-1, -2, 0, (-2, -1), None])
 @pytest.mark.parametrize("correction", [True, False])
 @pytest.mark.parametrize("keepdim", [True, False])
-@pytest.mark.parametrize("use_legacy", [True, False])
-def test_std(device, batch_size, h, w, dim, correction, keepdim, use_legacy):
+def test_std(device, batch_size, h, w, dim, correction, keepdim):
     torch.manual_seed(0)
 
     torch_input_tensor = torch.randn((batch_size, h, w), dtype=torch.bfloat16)
@@ -27,7 +26,7 @@ def test_std(device, batch_size, h, w, dim, correction, keepdim, use_legacy):
 
     input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, device=device)
 
-    output_tensor = ttnn.std(input_tensor, dim=dim, keepdim=keepdim, correction=correction, use_legacy=use_legacy)
+    output_tensor = ttnn.std(input_tensor, dim=dim, keepdim=keepdim, correction=correction)
     output_tensor = ttnn.to_layout(output_tensor, ttnn.TILE_LAYOUT)
     output_tensor = ttnn.from_device(output_tensor)
 
@@ -38,10 +37,7 @@ def test_std(device, batch_size, h, w, dim, correction, keepdim, use_legacy):
     atol = 0.01
     frobenius = 0.005
     pcc = 0.9999
-    if use_legacy:
-        # Legacy implementation is even less accurate.
-        pcc = 0.975
-    elif dim == (-2, -1):
+    if dim == (-2, -1):
         # For 2D reduction, all output values are close to 1, and we're using bfloat16,
         # so a rounding error of even 1 ULP impacts PCC.
         # ATOL/RTOL/Frobenius should catch any significant errors.
@@ -68,8 +64,7 @@ def test_std(device, batch_size, h, w, dim, correction, keepdim, use_legacy):
 @pytest.mark.parametrize("dim", [None, [], -1, -2, (-2, -1)])
 @pytest.mark.parametrize("keepdim", [True])
 @pytest.mark.parametrize("correction", [True, False])
-@pytest.mark.parametrize("use_legacy", [True, False])
-def test_var(device, batch_size, h, w, dim, keepdim, correction, use_legacy):
+def test_var(device, batch_size, h, w, dim, keepdim, correction):
     torch.manual_seed(0)
 
     torch_input_tensor = torch.randn((batch_size, h, w), dtype=torch.bfloat16)
@@ -77,7 +72,7 @@ def test_var(device, batch_size, h, w, dim, keepdim, correction, use_legacy):
 
     input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, device=device)
 
-    output_tensor = ttnn.var(input_tensor, dim=dim, keepdim=keepdim, correction=correction, use_legacy=use_legacy)
+    output_tensor = ttnn.var(input_tensor, dim=dim, keepdim=keepdim, correction=correction)
     output_tensor = ttnn.to_layout(output_tensor, ttnn.TILE_LAYOUT)
     output_tensor = ttnn.from_device(output_tensor)
 
@@ -90,11 +85,6 @@ def test_var(device, batch_size, h, w, dim, keepdim, correction, use_legacy):
     atol = 0.01
     pcc = 0.99999
     frobenius = 0.007
-
-    if use_legacy:
-        # Legacy implementation is less accurate.
-        pcc = 0.993
-        frobenius = 0.0085
 
     assert_numeric_metrics(
         torch_output_tensor,
@@ -150,7 +140,7 @@ def test_var_fp32_translation_invariance(device, dim, offset, scalar):
     torch_ref = torch.var((torch_input * scalar).to(torch.float64), dim=dim, keepdim=True, correction=correction)
 
     tt_in = ttnn.from_torch(torch_input, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
-    tt_out = ttnn.var(tt_in, dim=dim, scalar=scalar, keepdim=True, correction=correction, use_legacy=False)
+    tt_out = ttnn.var(tt_in, dim=dim, scalar=scalar, keepdim=True, correction=correction)
     actual = ttnn.to_torch(ttnn.from_device(tt_out))
 
     # Tolerances are tight: with the precision-preserving unpacker path the answer is
@@ -201,7 +191,7 @@ def test_var_fp32_doscale_wt_gt_1(device, scalar, N):
     torch_ref = torch.var((torch_input * scalar).to(torch.float64), dim=-1, keepdim=True, correction=correction)
 
     tt_in = ttnn.from_torch(torch_input, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
-    tt_out = ttnn.var(tt_in, dim=-1, keepdim=True, correction=correction, scalar=scalar, use_legacy=False)
+    tt_out = ttnn.var(tt_in, dim=-1, keepdim=True, correction=correction, scalar=scalar)
     actual = ttnn.to_torch(ttnn.from_device(tt_out))
 
     # Tolerances tighter than the BF16 floor: the FPU mul + cb_scaled UnpackToDest path
@@ -944,8 +934,7 @@ def test_run_reduce_sum_h_after_max_pool(device, input_shape, kernel_size):
         ([32, 32, 32, 0], False, 3, "var"),
     ],
 )
-@pytest.mark.parametrize("use_legacy", [True, False])
-def test_torch_compatibility(device, tensor_shape, keepdim, dim, op, use_legacy):
+def test_torch_compatibility(device, tensor_shape, keepdim, dim, op):
     """
     Test the compatibility of the torch and ttnn output for the given operation and different
     tensor shapes, keepdim, and dim values.
@@ -954,8 +943,6 @@ def test_torch_compatibility(device, tensor_shape, keepdim, dim, op, use_legacy)
     Note: We do not enforce the same exception type or message.
     """
     torch.manual_seed(42)
-    if op not in ("std", "var") and use_legacy:
-        pytest.skip("use_legacy only applies to std and var")
 
     rank = len(tensor_shape)
 
@@ -964,10 +951,6 @@ def test_torch_compatibility(device, tensor_shape, keepdim, dim, op, use_legacy)
     ttnn_tensor = ttnn.fill_implicit_tile_padding(ttnn_tensor, TEST_PADDING_VALUE)
 
     torch_op, ttnn_op = getattr(torch, op), getattr(ttnn, op)
-
-    ttnn_extra_kwargs = {}
-    if op in ("std", "var"):
-        ttnn_extra_kwargs["use_legacy"] = use_legacy
 
     # Run on both and flag exceptions
     torch_errored = False
@@ -978,7 +961,7 @@ def test_torch_compatibility(device, tensor_shape, keepdim, dim, op, use_legacy)
 
     ttnn_errored = False
     try:
-        ttnn_result = ttnn_op(ttnn_tensor, dim=dim, keepdim=keepdim, **ttnn_extra_kwargs)
+        ttnn_result = ttnn_op(ttnn_tensor, dim=dim, keepdim=keepdim)
     except RuntimeError:
         ttnn_errored = True
 

--- a/ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.cpp
+++ b/ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.cpp
@@ -9,8 +9,6 @@
 #include "ttnn/operations/data_movement/fill_pad/fill_pad.hpp"
 #include "ttnn/operations/data_movement/transpose/transpose.hpp"
 #include "ttnn/operations/data_movement/slice/slice.hpp"
-#include "ttnn/operations/eltwise/unary/unary.hpp"
-#include "ttnn/operations/eltwise/binary/binary_composite.hpp"
 #include "ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp"
 #include "ttnn/operations/reduction/generic/device/reduce_op.hpp"
 #include "ttnn/operations/reduction/reduction_common/reduction_common.hpp"
@@ -37,8 +35,7 @@ Tensor reduce(
     const std::optional<DeviceComputeKernelConfig>& compute_kernel_config = std::nullopt,
     float scalar = 1.0f,
     bool correction = true,
-    const std::optional<CoreRangeSet>& sub_core_grids = std::nullopt,
-    bool use_legacy = false);
+    const std::optional<CoreRangeSet>& sub_core_grids = std::nullopt);
 
 // input_shape has original shape while output_shape has reduction applied and last 2 dims padded.
 // Need to get slice parameters based on the minimum of the two shapes.
@@ -317,8 +314,7 @@ static Tensor std_var_impl(
     float scalar,
     const ttnn::SmallVector<int>& non_height_width_dims,
     bool correction,
-    const std::optional<CoreRangeSet>& sub_core_grids,
-    bool use_legacy) {
+    const std::optional<CoreRangeSet>& sub_core_grids) {
     auto input_shape = input_tensor_arg.logical_shape();
     auto rank = input_shape.size();
     auto memory_config = memory_config_arg.value_or(input_tensor_arg.memory_config());
@@ -356,71 +352,6 @@ static Tensor std_var_impl(
     // This could fail if e.g. there is only one element across the reduction dimensions and correction is true.
     uint64_t divisor = correction ? (reduced_volume - 1) : reduced_volume;
     TT_FATAL(divisor > 0, "Reduction is performed on too few elements, yielding divisor of {}", divisor);
-
-    if (use_legacy) {
-        // Classical two-pass variance via the identity  Var = E[X^2] - (E[X])^2.
-        // Let N = reduced_volume,  D = divisor  (D = N-1 when correction=true, D = N otherwise).
-        // The target formula is:
-        //   Var = Sum(X^2) / D  -  Sum(X)^2 / (N * D)
-        // This gives population variance when D = N, and sample variance
-        // (Bessel-corrected) when D = N-1.
-        // We bake the correction into the reduction scalars to minimize number of
-        // operations. The code computes:
-        //   mean_sq_tensor = Sum(X^2) * sq_scalar          ... (A)
-        //   mean_tensor    = Sum(X)   * mean_scalar         ... (B)
-        //   output         = (A) - (B)^2
-        //                  = Sum(X^2) * sq_scalar  -  Sum(X)^2 * mean_scalar^2
-        //
-        // Matching term-by-term against the target formula:
-        //   sq_scalar       = 1 / D
-        //   mean_scalar^2   = 1 / (N * D)   =>   mean_scalar = 1 / sqrt(N * D)
-        //
-        // For correction=false  (D = N):
-        //   sq_scalar   = 1/N and mean_scalar = 1/sqrt(N^2) = 1/N
-        //   (both equal -> population variance)
-        // For correction=true   (D = N-1):
-        //   sq_scalar   = 1/(N-1)
-        //   mean_scalar = 1/sqrt(N*(N-1))
-        //   output      = Sum(X^2)/(N-1) - Sum(X)^2/(N*(N-1))  (sample variance)
-        float N = static_cast<float>(reduced_volume);
-        float D = static_cast<float>(divisor);
-        float sq_scalar = scalar / D;
-        float mean_scalar = scalar / std::sqrt(N * D);
-
-        auto mean_tensor = reduce_impl<reduction_common::ReduceType::Sum>(
-            input_tensor_arg,
-            dim,
-            keepdim,
-            memory_config_arg,
-            compute_kernel_config,
-            mean_scalar,
-            non_height_width_dims,
-            sub_core_grids);
-
-        auto mean_square_tensor = reduce_impl<reduction_common::ReduceType::Sum>(
-            ttnn::pow(input_tensor_arg, 2.0f, memory_config),
-            dim,
-            keepdim,
-            memory_config_arg,
-            compute_kernel_config,
-            sq_scalar,
-            non_height_width_dims,
-            sub_core_grids);
-
-        Tensor output_tensor = ttnn::subtract(
-            mean_square_tensor, ttnn::pow(mean_tensor, 2.0f, memory_config), std::nullopt, memory_config);
-
-        // The two-pass formula can yield slightly negative values due to
-        // catastrophic cancellation, especially in low-precision (bfloat16).
-        // Clamp to zero because negative variance is meaningless, and sqrt of
-        // a negative value would produce inf when computing std.
-        output_tensor = ttnn::relu(output_tensor, memory_config);
-
-        if constexpr (reduce_type == reduction_common::ReduceType::Std) {
-            output_tensor = ttnn::sqrt(output_tensor, false, memory_config);
-        }
-        return output_tensor;
-    }
 
     // Welford single-pass algorithm
     bool single_h = (dim.size() == 1 && dim[0] == rank - 2);
@@ -567,8 +498,7 @@ Tensor reduce(
     const std::optional<DeviceComputeKernelConfig>& compute_kernel_config,
     float scalar,
     bool correction,
-    const std::optional<CoreRangeSet>& sub_core_grids,
-    bool use_legacy) {
+    const std::optional<CoreRangeSet>& sub_core_grids) {
     ttnn::SmallVector<int> dim = reduction_common::generate_reduce_dim(input_tensor_arg, dim_arg);
     float pad_value = get_pad_value(reduce_type, input_tensor_arg.dtype());
     // TODO: generalize to support all types, parameters, and formats. Issue #18566
@@ -585,8 +515,7 @@ Tensor reduce(
             scalar,
             non_height_width_dims,
             correction,
-            sub_core_grids,
-            use_legacy);
+            sub_core_grids);
     }
 
     bool is_tiled = input_tensor_arg.layout() == TILE_LAYOUT;
@@ -800,7 +729,7 @@ Tensor std(
     float scalar,
     bool correction,
     const std::optional<CoreRangeSet>& sub_core_grids,
-    bool use_legacy) {
+    bool /*use_legacy - deprecated and non-functional, kept for API compatibility*/) {
     return operations::reduction::reduce<reduction_common::ReduceType::Std>(
         input_tensor_arg,
         dim_arg,
@@ -809,8 +738,7 @@ Tensor std(
         compute_kernel_config,
         scalar,
         correction,
-        sub_core_grids,
-        use_legacy);
+        sub_core_grids);
 }
 
 Tensor var(
@@ -822,7 +750,7 @@ Tensor var(
     float scalar,
     bool correction,
     const std::optional<CoreRangeSet>& sub_core_grids,
-    bool use_legacy) {
+    bool /*use_legacy - deprecated and non-functional, kept for API compatibility*/) {
     return operations::reduction::reduce<reduction_common::ReduceType::Var>(
         input_tensor_arg,
         dim_arg,
@@ -831,8 +759,7 @@ Tensor var(
         compute_kernel_config,
         scalar,
         correction,
-        sub_core_grids,
-        use_legacy);
+        sub_core_grids);
 }
 
 }  // namespace ttnn

--- a/ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.hpp
+++ b/ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.hpp
@@ -67,6 +67,8 @@ Tensor min(
     bool correction = true,
     const std::optional<CoreRangeSet>& sub_core_grids = std::nullopt);
 
+// use_legacy is deprecated and non-functional: the Welford implementation is always
+// used. The parameter is kept only for API compatibility and will be removed.
 Tensor std(
     const Tensor& input_tensor_arg,
     const std::optional<std::variant<int, int64_t, SmallVector<int>>>& dim_arg = std::nullopt,
@@ -78,6 +80,8 @@ Tensor std(
     const std::optional<CoreRangeSet>& sub_core_grids = std::nullopt,
     bool use_legacy = false);
 
+// use_legacy is deprecated and non-functional: the Welford implementation is always
+// used. The parameter is kept only for API compatibility and will be removed.
 Tensor var(
     const Tensor& input_tensor_arg,
     const std::optional<std::variant<int, int64_t, SmallVector<int>>>& dim_arg = std::nullopt,

--- a/ttnn/cpp/ttnn/operations/reduction/generic/std_var_reductions_nanobind.cpp
+++ b/ttnn/cpp/ttnn/operations/reduction/generic/std_var_reductions_nanobind.cpp
@@ -34,7 +34,7 @@ static std::string get_std_var_reduction_doc(const char* op_name, const char* qu
             scalar (float, optional): A scaling factor to be applied to the input tensor. Defaults to `1.0`.
             correction (bool, optional): Whether to apply Bessel's correction (i.e. N-1). Defaults to `True`.
             sub_core_grids (ttnn.CoreRangeSet, optional): Subcore grids to use for the operation. Defaults to `None`, which will use all cores.
-            use_legacy (bool, optional): Deprecated and non-functional. The numerically stable Welford algorithm is always used. This argument is ignored and will be removed in 30 days (after July 10, 2026). Defaults to `False`.
+            use_legacy (bool, optional): Deprecated and non-functional. The numerically stable Welford algorithm is always used. This argument is ignored and will be removed at some point. Defaults to False.
 
         Returns:
             ttnn.Tensor: the output tensor.

--- a/ttnn/cpp/ttnn/operations/reduction/generic/std_var_reductions_nanobind.cpp
+++ b/ttnn/cpp/ttnn/operations/reduction/generic/std_var_reductions_nanobind.cpp
@@ -34,7 +34,7 @@ static std::string get_std_var_reduction_doc(const char* op_name, const char* qu
             scalar (float, optional): A scaling factor to be applied to the input tensor. Defaults to `1.0`.
             correction (bool, optional): Whether to apply Bessel's correction (i.e. N-1). Defaults to `True`.
             sub_core_grids (ttnn.CoreRangeSet, optional): Subcore grids to use for the operation. Defaults to `None`, which will use all cores.
-            use_legacy (bool, optional): When True, uses the legacy two-pass implementation. When False, uses the numerically stable Welford algorithm. Defaults to `False`.
+            use_legacy (bool, optional): Deprecated and non-functional. The numerically stable Welford algorithm is always used. This argument is ignored and will be removed in 30 days (after July 10, 2026). Defaults to `False`.
 
         Returns:
             ttnn.Tensor: the output tensor.

